### PR TITLE
DESCW-2771 Adds tests for Entity/Alert class

### PR DESCRIPTION
**Pull request by robrien-bcgov**
[DESCW-2771](https://citz-gdx.atlassian.net/browse/DESCW-2825)

- edits setFailures to handle edge case where $failure is negative.
- tests all trivial setter/getter methods in the same test case
- separates fromXML() Exception and Alert test cases.
image

### CODE REVIEW

File: tests/AlertTest.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    /**
+     * Get a valid XML string for testing
+     *
+     * @return string
+     */
+    private function getValidXmlString(): string
+    {
+        return '<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mhaswell-bcgov) [inline tests/AlertTest.php:160]: We could use `tests/Socket/complete-alert.xml` here which is what a valid alert will look like in production.

File: src/Entity/Alert.php [hunk 2]
```diff
@@ -194,22 +200,24 @@ public function setSuccess( bool $success ): void
...
-        $alert = new Alert();
-        $alert->setId($identifier);
-        $alert->setBody($xml->asXML());
-        $alert->setReceived(new DateTime());
+
+        $alert = new self();
+        // Avoiding unnecessary calls to getters and setters
+        $alert->id = $identifier;
+        $alert->body = $xml->asXML();
+        $alert->received = new DateTime();
```
Comment (mhaswell-bcgov) [inline src/Entity/Alert.php:220]: I think it's slightly better to use the setters here because they have some extra checks in some cases (like setFailures() forcing the value to be >0).

**Review by mhaswell-bcgov — APPROVED** [review]

File: src/Entity/Alert.php [hunk 3]
```diff
@@ -194,19 +200,20 @@ public function setSuccess( bool $success ): void
...
+
+        $err = 'Invalid XML: "identifier" field is required and must not be empty.';
+        if ('' === $identifier) {
+            throw new \InvalidArgumentException($err);
         }
-        
-        $alert = new Alert();
+
+        $alert = new self();
         $alert->setId($identifier);
```
Comment (robrien-bcgov) [inline src/Entity/Alert.php:217]: reverted to using setters. this lets us leverage any error-checking logic in the getter/setters.

File: tests/AlertTest.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+     * and body content.
+     *
+     * @return void
+     *
+     * @throws \Exception If there's an error parsing the XML or creating
+     *                    the Alert object.
+     */
+    public function testFromXmlWithValidData(): void
+    {
+        $xmlString = file_get_contents(self::XML_TEST_FILE);
```
Comment (robrien-bcgov) [inline tests/AlertTest.php:115]: grabs a complete alert XML from the test file

**Review by mhaswell-bcgov — APPROVED** [review]
